server/bin/pbench-server-put-shim-001.py

```diff
@@ -0,0 +1,166 @@
+
+    receive_dir_prefix = config.get("pbench-server", "pbench-receive-dir-prefix")
+    if not receive_dir_prefix:
+        logger.error(
+            "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
+        )
+        return None
+
+    receive_dir = Path(f"{receive_dir_prefix}-002").resolve()
+    if not receive_dir.is_dir():
+        logger.error("Failed: {} does not exist, or is not a directory", receive_dir)
+        return None
+
+    return receive_dir
+
+
+def process_tb(
+    config: PbenchServerConfig, logger: Logger, receive_dir: Path
+) -> Results:
+    """process_tb - receives a tar ball and copies it to the remote server
+                    using the server's HTTP PUT method
+
+        Args -
+            config -- PbenchServer config object
+            logger -- logger object to use when emitting log entries during
+                      operation
+            receive_dir -- directory where tarballs are received from agent
+
+    """
+
+    # Check for results that are ready for processing: version 002 agents
+    # upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+    # after they are done with MD5 checking so that's what we look for.
+    list_check = glob.glob(
+        os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
+    )
+
+    logger.info("{}", config.TS)
+    list_check.sort()
+    nstatus = ""
+
+    ntotal = ntbs = 0
+
+    token = config.get("pbench-server", "put-token")
+
+    for tbmd5 in list_check:
+        ntotal += 1
+
+        # Extracts full pathname of tarball from the name of hashfile
+        # by trimming .md5 suffix
+        tb = Path(tbmd5[0 : -len(".md5")])
+        tbmd5 = Path(tbmd5)
+        tbdir = tb.parent
+        controller = tbdir.name
+
+        try:
+            tarball_len, tarball_md5 = md5sum(tb)
+            crt = CopyResultTb(
+                controller, tb, tarball_len, tarball_md5, config, logger,
+            )
+            crt.copy_result_tb(token)
+        except Exception as e:
+            logger.error("{}: Unexpected Error: {}", config.TS, e)
+            continue
+
+        try:
+            os.remove(tbmd5)
+            os.remove(tb)
+        except Exception as exc:
+            logger.error(
+                "{}: Warning: cleanup of successful copy operation failed: '{}'",
+                config.TS,
+                exc,
+            )
+
+        ntbs += 1
+
+        nstatus = f"{nstatus}{config.TS}: processed {tb}\n"
+        logger.info(f"{tb.name}: OK")
+
+    return Results(nstatus, ntotal, ntbs)
+
+
+def main(cfg_name: PbenchServerConfig) -> int:
+    if not cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable or use --config <file> on the"
+            " command line",
+            file=sys.stderr,
+        )
+        return 2
+
+    try:
+        config = PbenchServerConfig(cfg_name)
```

Comment:
Correct me if I am wrong but according to the type-hints the `cfg_name` parameter is already of type `PbenchServerConfig` why do we need to reconvert it to `PbenchServerConfig`? I think in the type hints the cfg_name parameter should have been `str`/`Path` maybe?